phobos/blender/reserved_keys.py

```diff
@@ -6,5 +6,6 @@
 INERTIAL_KEYS = ["inertia", "mass", "origin", "joint_relative_origin"]
 MOTOR_KEYS = ["name", "type", "joint", "maxSpeed", "maxValue", "maxEffort", "minValue"]
 INTERFACE_KEYS = ["name", "type", "direction", "parent", "origin"]
-INTERNAL_KEYS = ["phobostype", "phobosmatrixinfo"]
-ANNOTATION_KEYS = ["$name", "$category", "$include_parent", "$include_transform", "$transform"]
\ No newline at end of file
+INTERNAL_KEYS = ["phobostype", "phobosmatrixinfo","submechanism/id"]
```

Comment:
instead of listing submechanism/id here you could add submechanism to the joint, and make it a reserved key of joints. Then you give it the submechanism object as value, that way changes in the submechanism object name will update here as well.